web/src/hotkey.js

```diff
@@ -183,7 +183,14 @@ export function get_keydown_hotkey(e) {
 
     const isCmdOrCtrl = common.has_mac_keyboard() ? e.metaKey : e.ctrlKey;
     if (isCmdOrCtrl && !e.shiftKey) {
-        hotkey = keydown_cmd_or_ctrl_mappings[e.which];
+        // On Mac, we treat Cmd as equivalent to Ctrl on Windows/Linux, and thus the ascii code
+        // for Cmd+return is the same as that for Ctrl+enter. so we check if the key pressed is
+        // cmd+return we get the key mapping from the keydown_ctrl_mappings.
+        if (e.which === 13) {
+            hotkey = keydown_ctrl_mappings[e.which];
```

Comment:
Shouldn't we just move the value from `keydown_ctrl_mappings[13]` to `keydown_cmd_or_ctrl_mappings`? It seems that `ctrl_enter` is used for both Ctrl and Cmd.

Reply:
I had considered doing the same thing before, and everything was working, but I couldn't figure out why the `ctrl_enter` mapping is not written inside `keydown_cmd_or_ctrl_mappings`. Additionally, having the `ctrl_enter` mapping in both `keydown_ctrl_mappings` and `keydown_cmd_or_ctrl_mappings` would be duplicative, so I wrote this if-else condition. Please let me know if you would like me to revise the approach and include the mapping of `ctrl_enter` in `keydown_cmd_or_ctrl_mappings` instead of using the current method.

Reply:
I'm not familiar with the historical context, but I believe shifting `ctrl_enter` to `keydown_cmd_or_ctrl_mappings` makes sense logically. However, it might require additional testing to ensure everything functions correctly.

Reply:
> I couldn't figure out why the `ctrl_enter` mapping is not written inside `keydown_cmd_or_ctrl_mappings`.

If I'm understanding correctly, the only mapping that needs to remain in `keydown_ctrl_mappings` is the escape one (`[`), because [at least in Firefox on MacOS](https://support.mozilla.org/en-US/kb/keyboard-shortcuts-perform-firefox-tasks-quickly), `Cmd` + `[` moves back through the browsing history. The same is true [in Chrome](https://support.google.com/chrome/answer/157179?hl=en&co=GENIE.Platform%3DDesktop#zippy=%2Ctab-window-shortcuts) and [Safari](https://support.apple.com/guide/safari/keyboard-and-other-shortcuts-cpsh003/mac).

Reply:
@shameondev @karlstolley I have changed the mapping of `ctrl_enter` to `keydown_cmd_or_ctrl_mappings` and also fixed the case for `cmd+return` and `ctrl+enter` to send the message on preview, could you please review the PR, thanks :)